sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ClientEncryptionPolicy.java

```diff
@@ -88,28 +104,44 @@ public int getPolicyFormatVersion() {
         return policyFormatVersion;
     }
 
-    void validatePartitionKeyPathsAreNotEncrypted(List<List<String>> partitionKeyPathTokens) {
+    /**
+     * Ensures that partition key paths specified in the client encryption policy for encryption are encrypted using Deterministic encryption algorithm.
+     * @param partitionKeyPathTokens Tokens corresponding to validated partition key.
+     */
+    void validatePartitionKeyPathsIfEncrypted(List<List<String>> partitionKeyPathTokens) {
         checkNotNull(partitionKeyPathTokens, "partitionKeyPathTokens cannot be null");
-        List<String> propertiesToEncrypt =
-            this.includedPaths.stream().map(clientEncryptionIncludedPath -> clientEncryptionIncludedPath.getPath().substring(1)).collect(Collectors.toList());
+
 
         for (List<String> tokensInPath : partitionKeyPathTokens) {
             checkNotNull(tokensInPath);
             if (tokensInPath.size() > 0) {
                 String topLevelToken = tokensInPath.get(0);
-                if (propertiesToEncrypt.contains(topLevelToken)) {
+
+                // paths in included paths start with "/". Get the ClientEncryptionIncludedPath and validate.
+                List<ClientEncryptionIncludedPath> encrypterPartitionKeyPath =
+                    this.includedPaths.stream().filter(clientEncryptionIncludedPath -> clientEncryptionIncludedPath.getPath().substring(1).equals(topLevelToken)).collect(Collectors.toList());
+
+                if (encrypterPartitionKeyPath.size() >0) {
+                    if (this.policyFormatVersion < 2) {
+                        throw new IllegalArgumentException(String.format("Path %s which is part of the partition key " +
+                            "cannot be encrypted" +
+                            " with PolicyFormatVersion %s", topLevelToken, policyFormatVersion));
+                    }
+
+                    // for the ClientEncryptionIncludedPath found check the encryption type.
+                    if (!encrypterPartitionKeyPath.stream().map(encrypter -> encrypter.getEncryptionType()).findFirst().orElse(null).equals("Deterministic"))
```

Comment:
why are we not using enum for checking the encryption type? Using literal string is not a good practice.

Reply:
I have fixed it now.